packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -677,14 +677,19 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     }
 
     if (!(ev.altKey || ev.metaKey) && (isUp || isDown)) {
-      const state = this.state;
+      this.setState(state => {
+        if (state.menuProps && !state.menuProps.shouldFocusOnMount) {
+          return { menuProps: { ...state.menuProps, shouldFocusOnMount: true } };
+        }
+        return state;
+      });
```

Comment:
I thought this would just be in the same location (line 687) without the `if` check. Did you have to move it?

Reply:
I'm a bit confused, it looks like it is in the same location. Which bit of code are you talking about?

Reply:
It's a bit more apparent if you look at the [commit](https://github.com/OfficeDev/office-ui-fabric-react/pull/8741/commits/bb3dc9ac5f333ccee2861cd81f20c53d94d1311c). Just wanted to check on the reasoning for moving it down then back up.